### PR TITLE
New API for Vert.x consul client 

### LINKED ISSUE #61: New API
The new client API, which will be much closer to the original golang interface provided by Hashicorp. The current API will be marked as deprecated and deleted in the future.

https://github.com/vert-x3/wiki/wiki/Vert.x-Roadmap

package: io.vertx.consul.v1
dev. branch: v1

### PR DESCRIPTION
**Pull request by Sammers21**
fixes #61

### CODE REVIEW

File: src/main/java/io/vertx/ext/consul/v1/Catalog.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package io.vertx.ext.consul.v1;
```
Comment (ruslansennov) [inline src/main/java/io/vertx/ext/consul/v1/Catalog.java]: @vietj for the new API we propose package `io.vertx.consul.v1`, without `ext`. Is it OK?

Comment (vietj) [reply src/main/java/io/vertx/ext/consul/v1/Catalog.java]: without ext is better, why v1 ?

Comment (ruslansennov) [reply src/main/java/io/vertx/ext/consul/v1/Catalog.java]: This is the official part of the Consul API versioning, even in packages

**Comment by vietj** [comment]
what is the status of this PR ? @Sammers21 @ruslansennov 

we are currently adding future to the API and it would be good to know whether we do it on the current API in master or on this new API

**Comment by ruslansennov** [comment]
Hi @vietj 
Unfortunately, we have to do this on the current API in master. We don't have time to make such huge change in the near future.

**Comment by vietj** [comment]
@ruslansennov you mean you don't have time to continue contributing this PR ?

**Comment by ruslansennov** [comment]
@vietj yes we both

**Comment by vietj** [comment]
ok

**Pull request closed without merging**